base/generator_eltype.jl

```diff
@@ -0,0 +1,28 @@
+function eltype(::Type{Generator{A, typeof(identity)}}) where {A}
+    eltype(A)
+end
```

Comment:
TIL that `x for x in A` is noticed by the parser, and produces this type:
```julia
julia> (x for x in 1:3)
Base.Generator{UnitRange{Int64}, typeof(identity)}(identity, 1:3)

julia> (identity(x) for x in 1:3)
Base.Generator{UnitRange{Int64}, var"#13#14"}(var"#13#14"(), 1:3)
```
(Was true on Julia 1.6, not true on 1.0.)

I wonder what would break if this instead returned A, i.e. we made this `(x for x in 1:3) === 1:3`.

Reply:
Yeah, I discovered the same thing while working on the PR, and wonder the same thing, too.

Reply:
Actually, I know what would break. `map(f, x)` is implemented as something like `collect(Iterators.map(f, x))`. This relies on the fact that `Generator` has `EltypeUnknown`.

Reply:
Lots of code assumes that, e.g., `map(identity, x)` will return a collection with more precise `eltype` than `x` has. This is what relies on `Generator` having `EltypeUnknown`.